packages/server/src/fhir/lookups/reference.ts

```diff
@@ -99,16 +94,16 @@ function getSearchReferences(resource: WithId<Resource>): ReferenceRow[] {
       if (value.type === PropertyType.Reference && value.value.reference) {
         const targetId = resolveId(value.value);
         if (targetId && isUUID(targetId)) {
-          addSearchReferenceResult(result, resource, searchParam, targetId);
+          addSearchReferenceResult(resultMap, resource, searchParam, targetId);
         }
       }
 
       if (isResource(value.value) && value.value.id && isUUID(value.value.id)) {
-        addSearchReferenceResult(result, resource, searchParam, value.value.id);
+        addSearchReferenceResult(resultMap, resource, searchParam, value.value.id);
       }
     }
   }
-  return Array.from(result.values());
+  result.push(...Array.from(resultMap.values()));
```

Comment:
```suggestion
  result.push(...resultMap.values());
```
This should avoid an intermediate array